### PR TITLE
85 implement detection of circular dependency

### LINKED ISSUE #85: Implement detection of circular dependency
All constructor dependencies are mandatory for bean to be created, thus those dependencies must be created first. It's possible to configure dependencies in a circular way. 
For example, **ServiceA** -> ServiceB -> ServiceC -> **ServiceA**, **->** character means dependency. In this case, no object could be instantiated first, because ServiceC is dependent on ServiceA which is the very first bean in the chain.
Such dependencies should be detected and corresponding self-explanatory exception should be thrown, also it's nice to have some additional logging that will explain better what happened.
Circular dependencies should be detected when all bean definitions are done, but before creation of the BringContainer object.
When a circular dependency is detected, it's important to collect the full circular path and show it to the user in the way: **ServiceA -> ServiceB -> ServiceC -> ServiceA**.
_**(Optional)**_ In case when short circular dependency is found in the long chain, it will be nice to show only the path for the short chain, rather than the full chain. For example in case ServiceA -> ServiceB -> **ServiceC-> ServiceD -> ServiceE -> ServiceC**, it would be nice to see only the path where circular dependency is found, but not the whole path.

### PR DESCRIPTION
**Pull request by golichenko02**

### CODE REVIEW

File: src/test/java/io/github/bobocodebreskul/context/registry/BeanDefinitionValidatorTest.java [hunk 1]
```diff
@@ -0,0 +1,430 @@
...
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
```
Comment (golichenko02) [inline src/test/java/io/github/bobocodebreskul/context/registry/BeanDefinitionValidatorTest.java]: no need for thid annotation, can be removed

Comment (golichenko02) [reply src/test/java/io/github/bobocodebreskul/context/registry/BeanDefinitionValidatorTest.java]: done

**Review by JoshukLite — APPROVED** [review]

File: src/main/java/io/github/bobocodebreskul/context/support/GeneralConstants.java [hunk 2]
```diff
@@ -9,5 +9,19 @@
  */
 @UtilityClass
 public class GeneralConstants {
-    public static final String EMPTY = "";
+    /**
```
Comment (mykolaFilimonov) [inline src/main/java/io/github/bobocodebreskul/context/support/GeneralConstants.java]: we not use this class anymore, lets delete it

**Review by mykolaFilimonov — APPROVED** [review]
NIT: fix 
LGTM

File: src/main/java/io/github/bobocodebreskul/context/support/GeneralConstants.java (hunk 2 above)
Comment (golichenko02) [reply src/main/java/io/github/bobocodebreskul/context/support/GeneralConstants.java]: done

**Pull request merged**